### ISSUE TITLE
📚 Course Topic Submission

### ISSUE THREAD

**Issue by leoxs22** [buildspace/buildspace-projects#51]
#### _Interaction with other contracts._

One of the most important things in blockchain, and also one of the most difficult, is to interact with other contracts.
How do you start working with another contract without having to use main net?
How can you test against this contract when you probably don't have the resources it requires

I think it would be awesome a course that:

- Gets the current balance of a wallet of some ERC20
- Grabs the coin price from Uniswap
- Gets the current APY of that coin in AAVE
- Deposit the token in AAVE (because we need to do some writing too 😝 )

And maybe end it with getting some data off-chain using some oracle

### Have You Taken a Course On This Topic Before
[Cryptozombies](https://cryptozombies.io/) but only covers the coding part, no idea how to test without having to deploy in mainnet

### Do You Want To Write This Course?

I can help, but I don't really have all the required knowledge for it 😢

**Comment by adilanchian** [comment]
Whats up! Thanks so much for submitting this :)

It sounds like you want to see the full deployment pipeline of this?

**Comment by leoxs22** [comment]
> Whats up! Thanks so much for submitting this :)
> 
> It sounds like you want to see the full deployment pipeline of this?


Deploying on production is too expensive for a course, but deploying on rinkeby and interacting with other contracts there would be awesome.

Now that I think about it, since the topic is interaction with other contracts, it could be fun if the students made contracts that interact with each other

**Comment by adilanchian** [comment]
Yess all of our projects take place on test nets :)

Ooo I like this like a contract calling another contract?

**Comment by shryasss** [comment]
I got another one, How about a project which includes the Inheritance of Smart Contracts. When building a production-grade application, we need to have in mind that Smart Contract, once deployed is going to be immutable and we need to used proper architecture so that suppose in future we decide to add features, our data stored on smart contracts shouldn't get affected.

I've been building a Crypto-Twitter sort of app, and I faced this issue while building it. I'd like to add this course, but I'm not very proficient with hardhat, so if I get help, I'll be very happy to construct this course.

Anyway, loved the courses which are already there, they got me started in Web3, Thank you :)
